framework/base/Security.php

```diff
@@ -456,14 +448,19 @@ public function generateRandomKey($length = 32)
             throw new InvalidParamException('First parameter ($length) must be greater than 0');
         }
 
-        // The recent LibreSSL RNGs are faster and better than /dev/urandom.
+        // The recent LibreSSL RNGs are faster and likely better than /dev/urandom.
         // Parse OPENSSL_VERSION_TEXT because OPENSSL_VERSION_NUMBER is no use for LibreSSL.
         // https://bugs.php.net/bug.php?id=71143
-        if ($this->_randomSource === self::SOURCE_LIBRE_SSL
-            || ($this->_randomSource === null
-                && defined('OPENSSL_VERSION_TEXT')
+        if ($this->_libreSSL === null) {
+            $this->_libreSSL = defined('OPENSSL_VERSION_TEXT')
                 && preg_match('{^LibreSSL (\d\d?)\.(\d\d?)\.(\d\d?)$}', OPENSSL_VERSION_TEXT, $matches)
-                && (10000 * $matches[1]) + (100 * $matches[2]) + $matches[3] >= 20105)
+                && (10000 * $matches[1]) + (100 * $matches[2]) + $matches[3] >= 20105;
+        }
+
+        // Since 5.4.0, openssl_random_pseudo_bytes() reads from CryptGenRandom on Windows instead
+        // of using OpenSSL library. Don't use OpenSSL on other platforms.
+        if ($this->_libreSSL || (DIRECTORY_SEPARATOR !== '/'
+                && substr_compare(PHP_OS, 'win', 0, 3, true) === 0 && PHP_VERSION_ID >= 50400)
```

Comment:
added `substr_compare()` to positively identify OS as windows.

keep `DIRECTORY_SEPARATOR !== '/'` as optimization for non-windows. the compiler evaluates it so it takes no runtime and prunes the string compare.